cpp/src/arrow/util/thread_pool_test.cc

```diff
@@ -123,6 +123,125 @@ class AddTester {
   std::vector<int> outs_;
 };
 
+template <typename T = arrow::detail::Empty>
```

Comment:
Add comments for this?

Reply:
These methods went away addressing the next comment.